src/renderers/shared/stack/reconciler/ReactCompositeComponent.js

```diff
@@ -1002,11 +955,14 @@ var ReactCompositeComponentMixin = {
     this._updateRenderedComponent(transaction, unmaskedContext);
 
     if (hasComponentDidUpdate) {
-      if (__DEV__) {
-        transaction.getReactMountReady().enqueue(
-          invokeComponentDidUpdateWithTimer.bind(this, prevProps, prevState, prevContext),
-          this
-        );
+      if (__DEV__ && debugID !== 0) {
+        transaction.getReactMountReady().enqueue(() => {
+          measureLifeCyclePerf(
+            () => inst.componentDidUpdate(prevProps, prevState, prevContext),
+            debugID,
+            'componentDidUpdate'
+          );
+        }, null);
       } else {
         transaction.getReactMountReady().enqueue(
           inst.componentDidUpdate.bind(inst, prevProps, prevState, prevContext),
```

Comment:
In all the other blocks, you mirrored character for character the two branches but didn't do it here, I think it would be valuable to update this code to look exactly the same as the perf branch